clusterloader2/pkg/state/namespaces.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// InstancesState represents state of object replicas.
+type InstancesState struct {
+	Number int32
+	Object api.Object
+}
+
+type instancesIdentifier struct {
+	basename   string
+	objectKind string
+	apiGroup   string
+}
+
+// namespaceState represents state of a single namespace.
+type namespaceState map[instancesIdentifier]*InstancesState
+
+// namespacesState repersents state of all unsed namespaces.
+type namespacesState struct {
+	namespaces map[string]namespaceState
+	lock       sync.RWMutex
+}
+
+func (ns *namespacesState) get(namespace string, basename string, objectType string, apiGroup string) (*InstancesState, bool) {
```

Comment:
Please pass one single parameter `instancesIdentifier` instead of passing the same 3 fields everywhere.

Reply:
done.